### ISSUE TITLE
Query fails with ConnectionBusy error

### ISSUE THREAD

**Issue by ElrohirGT** [karlseguin/pg.zig#55]
Hi! I've been using pg.zig (on Zig@0.13.0) and so far I'm loving it, but I can't seem to do two queries with the same connection obtained from the pool!

Here's the code snippet:
```zig
juwura.logInfo("Getting DB connection...").log();
    const conn = self.pool.acquire() catch |err| {
        juwura.logErr("Error in DB connection").err(err).log();
        r.setStatus(.internal_server_error);
        r.sendBody("NO DB CONNECTION AQUIRED") catch unreachable;
        return;
    };
    defer conn.release();
    juwura.logInfo("Connection aquired!").log();

    const pj_query = "INSERT INTO project (name, photo_url) VALUES ($1, $2) RETURNING *";
    const pj_params = .{ request.name, request.photo_url };

    juwura.logInfo("Creating project in DB...").string("query", pj_query).string("name", request.name).string("photo_url", request.photo_url).log();
    const pj_result = conn.query(pj_query, pj_params) catch |err| {
        juwura.logErr("Error in query").err(err).log();
        r.setStatus(.internal_server_error);
        r.sendBody("QUERY ERROR") catch unreachable;
        return;
    };

    const dataRow = (pj_result.next() catch unreachable) orelse unreachable;
    const id = dataRow.get(i32, 0);
    const name = dataRow.get([]u8, 1);
    const url = dataRow.get(?[]u8, 2);

    pj_result.deinit();
    juwura.logInfo("Project created!").log();

    const mem_query = "INSERT INTO project_member (project_id, user_id) VALUES ($1, $2)";
    const mem_params = .{ id, request.email };

    juwura.logInfo("Adding project creator to members...").string("query", mem_query).int("projectId", id).string("userEmail", request.email).log();
    const memberResult = conn.query(mem_query, mem_params) catch |err| {
        juwura.logErr("Error in query").err(err).log();
        r.setStatus(.internal_server_error);
        r.sendBody("QUERY ERROR") catch unreachable;
        return;
    };
    memberResult.deinit();
    juwura.logInfo("Creator added to member!").log();
```

And it fails on the second conn.query, with the `ConnectionBusy` error. Shouln't it be already free to execute another query of the pj_result.deinit() call?

I tried to check the examples but all of them only execute a single query. Any help would be greatly appreciated!

**Comment by karlseguin** [comment]
TL;DR change your first query to use `.row` instead of `.query`.


When using `query`, the result _has_ to be drained. You either need to iterate it until `next()` return `null` or you need to call `result.drain()`.

The reason `result.deinit()` doesn't "drain" is because `drain()` can fail and you can't use `try` in a defer. So, in a typical usage, I expect users to do:

```zig
var result = ...
defer result.deinit();
while (try result.next()) |row| { 
 ...
}
```

If `deinit` could fail, this becomes awkward (again, because you can't use `try` in a defer). So I optimized the API for this "common" case.

If you don't fully drain the result, then you need to call `drain()` and deal with the possible error.

You can also use `conn.row` in your first query instead of `conn.query`. You'll need to call `deinit()` on the row (and it can fail, wow, why did I make it so inconsistent!)

Explained here:
https://github.com/karlseguin/pg.zig?tab=readme-ov-file#drainresult-result-void

**Comment by ElrohirGT** [comment]
![image](https://github.com/user-attachments/assets/bc99d100-2988-4b4f-bafa-b920571b0725)

Thank you so much! You saved me.

I remember reading through that part when trying to debug the issue but it never clicked that I should be calling drain on a query! Totally my bad.

have a nice day!
